### ISSUE TITLE
3.1.0 does not build on FreeBSD, overrides sys/endian.h #defines

### ISSUE THREAD

**Issue by mandree** [AcademySoftwareFoundation/openexr#1110]
On FreeBSD 13.0 amd64, OpenEXR 3.1.0's new core library overrides the sys/endian.h macros in `internal_xdr.h`. I need to remove six #defines to make OpenEXR compile. I am not sure about NetBSD's or DragonflyBSD's header files, so not sure if that's appropriate for them:

```
--- src/lib/OpenEXRCore/internal_xdr.h.orig     2021-07-21 19:25:24 UTC
+++ src/lib/OpenEXRCore/internal_xdr.h
@@ -67,12 +67,6 @@
 #elif defined(__NetBSD__) || defined(__FreeBSD__) || defined(__DragonFly__)
 
 #  include <sys/endian.h>
-#  define be16toh(x) betoh16(x)
-#  define le16toh(x) letoh16(x)
-#  define be32toh(x) betoh32(x)
-#  define le32toh(x) letoh32(x)
-#  define be64toh(x) betoh64(x)
-#  define le64toh(x) letoh64(x)
 #  define EXR_HOST_IS_NOT_LITTLE_ENDIAN (__BYTE_ORDER != __LITTLE_ENDIAN)
 
 #else

```

**Comment by kdt3rd** [comment]
Yes, when I went to look up what was different, I had seen this pattern, but it must be ancient, all the BSD-ish O.S. provide similar sys/endian.h, so I've submitted a PR to just collapse all the BSDs into the one sys/endian.h

**Comment by mandree** [comment]
This was PR #1115 and got fixed with OpenEXR release v3.1.1. Thank you!
